### PR TITLE
Generic grid, director, rector and employee panels

### PR DESCRIPTION
**Pull request by MarceliJerzynski**
Linked issues: 

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/app/components/shared/grid/grid.component.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    for(let filter in this.columnHeader) {
+       this.columnFilters[filter + "Filter"] = this.columnHeader[filter];
+    }
+    this.dataSource = new CustomDataSource(this.restService);
+    this.dataSource.loadData(this.dataRestPath, this.pageInfo);
+
+
+  }
+
+  ngAfterViewInit() {
```
Comment (pietrzakkuba) [inline src/app/components/shared/grid/grid.component.ts:53]: `AfterViewInit` shoud be added to `export class GridComponent implements`

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:8080",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
[the remark this answers is not in the document]
Comment (MarceliJerzynski) [reply .vscode/launch.json:15]: > we probably don't need this vscode setup on github

Probably, I have no idea what this is

File: debug.log [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[1023/181730.485:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
+[1025/172254.617:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
+[1025/192155.124:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
+[1027/195848.666:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
+[1028/165908.511:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
+[1030/202328.335:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
+[1031/134926.084:ERROR:directory_reader_win.cc(43)] FindFirstFile: System nie mo�e odnale�� okre�lonej �cie�ki. (0x3)
```
[the remark this answers is not in the document]
Comment (MarceliJerzynski) [reply debug.log:7]: > don't need this too I think

Sorry bro, I will delete this immediately

**Review by pietrzakkuba — APPROVED** [review]
great work! 🎉

**Pull request merged**